rotkehlchen/assets/utils.py

```diff
@@ -33,7 +34,81 @@ def add_ethereum_token_to_db(token_data: EvmToken) -> EvmToken:
         data=token_data,
     )
     # This can, but should not raise UnknownAsset, DeserializationError
-    return EvmToken(token_data.identifier, form_with_incomplete_data=True)
+    return EvmToken(token_data.identifier)
+
+
+def _query_token_information(
+        evm_inquirer: 'EvmNodeInquirer',
+        evm_address: ChecksumEvmAddress,
+        name: Optional[str],
+        symbol: Optional[str],
+        decimals: Optional[int],
+        token_kind: EvmTokenKind,
+) -> tuple[Optional[str], Optional[str], Optional[int]]:
```

Comment:
The return type is wrong. They don't seem to be Optional. If `None` then you raise an error in line 59.

Reply:
`get_erc20_contract_info` can return `None` and any of the arguments provided can be `None`. That means that the returned values can be `None`